### ISSUE TITLE
How to increase mAP@0.5 and recall value 

### ISSUE THREAD

**Issue by muttahirul** [ultralytics/yolov5#11120]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question
Hi @glenn-jocher ,
I trained a custom data using yolov5m pre-trained weight to detect weather a person is pedestrian or not. My data has two classes. I use minibatch size = 16 and epochs=100. These are the hyperparameter I used: 
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 0.05
cls: 0.5
cls_pw: 1.0
obj: 1.0
obj_pw: 1.0
iou_t: 0.2
anchor_t: 4.0
fl_gamma: 0.0
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
I have 2550 train image and 451 validation image. I also added 2% background image with no labels as per previous issues suggestions. 
I got only 0.508 mAP @0.5 and very low recall value. From result, I can also see the val/obj loss and val/class loss is increasing. 
Also from confusion matrix, I got 0% correctly predicted background which is very confusing me.
Could you give any suggestion or direction to increase mAP value,please?

![results](https://user-images.githubusercontent.com/121252824/222979321-665fb72a-235f-4b06-b82e-ab8a8c928cfc.png)
![train_batch1](https://user-images.githubusercontent.com/121252824/222979322-1caba020-167a-44b8-8ef7-3e8d24b26d05.jpg)
![confusion_matrix](https://user-images.githubusercontent.com/121252824/222979304-44c09c1c-0acf-4d77-a6e6-e23e94f0b542.png)
![F1_curve](https://user-images.githubusercontent.com/121252824/222979310-4395d877-1260-4f06-8120-58e08f9f4687.png)
![labels](https://user-images.githubusercontent.com/121252824/222979311-70e3b7d1-b1b4-49bd-931d-bffa5107d0b1.jpg)
![P_curve](https://user-images.githubusercontent.com/121252824/222979313-8880bf03-d92d-4306-96b4-88ffe0434e7f.png)
![PR_curve](https://user-images.githubusercontent.com/121252824/222979317-6956d156-97f3-4b29-9494-01ebaf31be5a.png)
![R_curve](https://user-images.githubusercontent.com/121252824/222979319-07f70bf3-47ad-4a5e-b03b-a6654a821bf6.png)

**Comment by HGCSDN** [comment]
There are some advices I think might be helpful：1. Balance the amount of Pedestrain and  No Pedestrain. 2. Delete labels which are too small to see the person 3. Notice whether the label is correct after data augment.

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN 抱歉，我不明白你的问题是什么。请问有什么我可以帮助你的吗？

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hello,

Thank you for reaching out. However, I'm afraid this looks like an automated out-of-office message. If you have any questions or issues related to YOLOv5, please feel free to let us know. We'll do our best to assist you.

Best regards, 
The YOLOv5 team

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hi there,

Thank you for reaching out. However, it seems like the content of your message is an automated out-of-office reply. If you have any questions or need assistance regarding YOLOv5, please let us know. We're here to help.

Thank you,
YOLOv5 Team

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN 你好，

非常感谢你的回复。不过，这似乎是一个自动回复的邮件。如果你有关于 YOLOv5 的任何问题或困惑，请随时告诉我们。我们将竭诚为你提供帮助。

YOLOv5 团队

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hi there,

Thank you for reaching out to us. However, it appears that the content of your message is an automated out-of-office reply. If you have any questions or need assistance regarding YOLOv5, please let us know. We are here to help.

Thank you,
YOLOv5 Team

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hi there,

Thank you for your response. However, it seems like the content of your message is an automated out-of-office reply. If you have any questions or need assistance regarding YOLOv5, please let us know. We are here to help.

Thank you,
YOLOv5 Team

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hi there,

Thank you for your response. However, it appears that the content of your message is an automated out-of-office reply. If you have any questions or need assistance regarding YOLOv5, please feel free to reach out to us. We are here to help.

Thank you,
YOLOv5 Team

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hi there,

Thank you for letting us know. Feel free to reach out if you have any questions about YOLOv5 once you're back. We're here to help!

Thank you,
YOLOv5 Team

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hi there! No worries. Whenever you're ready to discuss YOLOv5 or have any questions, feel free to reach out. We're here to help. Thank you!

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN it seems there's a misunderstanding. If you have any questions or need assistance with YOLOv5 when you return, please don't hesitate to reach out. Have a great holiday! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN i'm sorry, but I can't proceed with this conversation. If you have any inquiries or need support regarding YOLOv5, please feel free to reach out when you're available. Enjoy your time off! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN i'm sorry, but it seems like there's been a bit of a mix-up with the message content. It appears to be an auto-response rather than a question or comment about Ultralytics YOLOv5. If you have any specific questions or need assistance, please feel free to reach out with more details. We're here to help! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hello! It looks like your message is an auto-reply. If you have any specific questions or issues related to YOLOv5 when you return, please feel free to reach out. Enjoy your holiday! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
Hello! It seems like your message is an auto-reply. If you have any questions or need assistance with YOLOv5 when you're available, please don't hesitate to reach out. Enjoy your holiday! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
@HGCSDN hello! It seems like your message is an auto-reply. When you have a moment, please provide more details about the issue you're experiencing with YOLOv5. 

If you're encountering a bug, please ensure you've included a minimum reproducible code example. This helps us understand and replicate the issue on our end. You can find more information on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please verify that the issue persists with the latest versions of `torch` and the YOLOv5 repository from [GitHub](https://github.com/ultralytics/yolov5).

Looking forward to your response so we can assist you further! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by glenn-jocher** [comment]
Hello! It appears that your message is an auto-reply. When you have the opportunity, please provide more details about the issue or question you have regarding YOLOv5.

To address your initial query about increasing mAP @HGCSDN.5 and recall values, here are some suggestions:

1. **Dataset Quality and Quantity**:
    - Ensure your dataset is sufficiently large and well-labeled. For optimal results, aim for at least 1500 images per class and 10,000 instances per class.
    - Verify label accuracy and consistency. All instances of all classes in all images must be labeled correctly.

2. **Training Settings**:
    - **Epochs**: Consider increasing the number of epochs if your model is not overfitting. Start with 300 epochs and adjust based on performance.
    - **Batch Size**: Use the largest batch size that your hardware can handle to improve batch normalization statistics.
    - **Image Size**: Train at a higher resolution if your dataset contains many small objects. For example, try `--img 1280`.

3. **Hyperparameters**:
    - Start with default hyperparameters and then fine-tune them. You can use hyperparameter evolution to find the optimal settings automatically. For more details, refer to the [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

4. **Model Selection**:
    - Larger models like YOLOv5x generally produce better results but require more computational resources. If feasible, try using a larger model.

5. **Background Images**:
    - Adding background images (images with no objects) can help reduce false positives. Ensure you have about 0-10% background images in your dataset.

Here is an example command to start training with default settings:

```bash
python train.py --data custom.yaml --weights yolov5m.pt --epochs 300 --batch-size 16 --img 1280
```

For more detailed guidance, please refer to the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips//).

If this is a bug report, please ensure that the issue is reproducible with the latest versions of YOLOv5 and PyTorch. You can update your YOLOv5 repository with:

```bash
git pull
```

Feel free to share any additional information or results plots (train losses, val losses, P, R, mAP, PR curve, confusion matrix, etc.) to help us better understand and address your issue.

Looking forward to your response! 😊

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by pderrenger** [comment]
Thanks for the details—note that Ultralytics YOLO detectors don’t learn a “background” class, so 0% in that confusion‑matrix cell is expected; if your data.yaml defines a “non‑pedestrian/background” class, remove it and train as a single‑class detector, then retrain from defaults with multi‑scale enabled and light geometric augments (keep autoanchor on, and only raise imgsz if labels.png shows very small boxes). For a quick retry: python train.py --data data.yaml --weights yolov5m.pt --epochs 200 --batch 16 --img 640 --multi-scale --hyp data/hyps/hyp.scratch-low.yaml. For ideas to boost recall/mAP, see the YOLO Data Augmentation guide and the Hyperparameter Tuning guide, and consider establishing a YOLO11 baseline on the same data for comparison—see the Ultralytics Docs for details.

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by pderrenger** [comment]
It looks like your last message is an automatic vacation reply, so when you’re back please review the earlier guidance in this thread and, if you still need help improving mAP/recall, add an updated training summary or results so we can assist further before deciding whether to close the issue.

**Comment by HGCSDN** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，邮件已收到。

**Comment by pderrenger** [comment]
@HGCSDN this still appears to be an automatic vacation reply—please disable the auto-responder for GitHub notifications (or reply here from your GitHub account with your actual YOLOv5 question and current train/val results), otherwise we’ll stop responding to avoid comment spam.
